python/lsst/daf/butler/dimension_universe_history.py

```diff
@@ -0,0 +1,51 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This software is dual licensed under the GNU General Public License and also
+# under a 3-clause BSD license. Recipients may choose which of these licenses
+# to use; please see the files gpl-3.0.txt and/or bsd_license.txt,
+# respectively.  If you choose the GPL option then the following text applies
+# (but note that there is still no warranty even if you opt for BSD instead):
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import json
+
+import yaml
+from lsst.resources import ResourcePath
+
+
+def load_historical_dimension_universe_json(universe_version: int) -> str:
+    """Load a specific version of the default dimension universe.
+
+    Parameters
+    ----------
+    universe_version : `int`
+        Version number of the universe to be loaded.
+
+    Returns
+    -------
+    universe : `str`
+        Dimension universe configuration encoded as a JSON string.
+    """
+    # This is used by daf_butler_migrate during dimension universe migrations.
```

Comment:
In general for `daf_butler_migrate` I try to avoid depending on `daf_butler` interfaces. `daf_butler_migrate` is very special in that it may need to work with different daf_butler versions, not just most recent (would be better to avoid this dependency completely, but some things are too expensive to reimplement).

Reply:
So would you prefer that I put the historical dimension universe YAMLs in daf_butler_migrate instead then?  (I actually slightly prefer that but Tim wanted them in daf_butler.)

Reply:
My argument for putting them in daf_butler is that I have been asked by people in the past about old universes and it seems easier to point them at a file in daf_butler than try to explain to them what daf_butler_migrate is. If the consensus is that people want them in daf_butler_migrate I will go along with that (and we have to remember to add them there when we create new ones.

Reply:
I think I like them better in daf_butler, but the code that reads them can go to daf_butler_migrate. I'm not sure that `load_historical_dimension_universe_json` is even generally useful in daf_butler.

Reply:
I thought it would be a good idea to have a stable function that daf_butler_migrate could call to insulate it from changes in daf_butler's package structure, but I don't feel that strongly about it -- I'll put the code in daf_butler_migrate and the YAMLs in daf_butler.